GoogleMapsComponents/Maps/BaseListableEntity.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.JSInterop;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GoogleMapsComponents.Maps
+{
+    public abstract class BaseListableEntity<T> : IDisposable, IJsObjectRef
+        where T: BaseListableEntityOptions
+    {
+        protected readonly JsObjectRef _jsObjectRef;
+
+        public readonly Dictionary<string, List<MapEventListener>> EventListeners;
+
+        public Guid Guid => _jsObjectRef.Guid;
+
+        internal BaseListableEntity(JsObjectRef jsObjectRef)
+        {
+            _jsObjectRef = jsObjectRef;
+            EventListeners = new Dictionary<string, List<MapEventListener>>();
+        }
+
+        public void Dispose()
+        {
+            foreach (string key in EventListeners.Keys)
+            {
+                //Probably superfluous...
+                if (EventListeners[key] != null)
+                {
+                    foreach (MapEventListener eventListener in EventListeners[key])
+                    {
+                        eventListener.Dispose();
+                    }
+
+                    EventListeners[key].Clear();
+                }
+            }
+
+            EventListeners.Clear();
+            _jsObjectRef.Dispose();
+        }
+
+        public Task<Map> GetMap()
+        {
+            return _jsObjectRef.InvokeAsync<Map>(
+                "getMap");
+        }
+
+
+
+        /// <summary>
+        /// Renders the mao entity on the specified map or panorama. 
+        /// If map is set to null, the map entity will be removed.
+        /// </summary>
+        /// <param name="map"></param>
+        public async Task SetMap(Map map)
+        {
+            await _jsObjectRef.InvokeAsync(
+                   "setMap",
+                   map);
+
+            //_map = map;
+        }
+
+        public async Task<MapEventListener> AddListener(string eventName, Action handler)
```

Comment:
Why methods are not virtual

Reply:
This methods are not virtual simply because BaseListableEntity is abstract only for the reason I want it to be inherited, but the methods are fully implemented -> those methos are exactly the same inside Marker and Circle, so I put them in a single piece of code inside a "BaseClass" but this class mean nothing as "Map object" it is to poor to represents something.
I have not found a more elegant way to do it.